packages/fxa-settings/src/pages/Pair/AuthTotp/index.tsx

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { FtlMsg } from 'fxa-react/lib/utils';
+// import { useFtlMsgResolver } from '../../../models/hooks';
+import { usePageViewEvent } from '../../../lib/metrics';
+// import { useAlertBar } from '../../models';
+import { TwoFactorAuthImage } from '../../../components/images';
+import CardHeader from '../../../components/CardHeader';
+import FormVerifyCode, {
+  FormAttributes,
+} from '../../../components/FormVerifyCode';
+import { MozServices } from '../../../lib/types';
+import { REACT_ENTRYPOINT } from '../../../constants';
+
+// --serviceName-- is the relying party
+
+export type AuthTotpProps = {
+  email: string;
+  serviceName?: MozServices;
+};
+
+export const viewName = 'auth-totp';
+
+const AuthTotp = ({
+  email,
+  serviceName,
+}: AuthTotpProps & RouteComponentProps) => {
+  usePageViewEvent(viewName, REACT_ENTRYPOINT);
+
+  const [code, setCode] = useState<string>('');
+  const [codeErrorMessage, setCodeErrorMessage] = useState<string>('');
+  // const alertBar = useAlertBar();
+  // const ftlMsgResolver = useFtlMsgResolver();
+
+  // These ftlids match those in `SigninTotpCode`
+  const formAttributes: FormAttributes = {
+    inputFtlId: 'signin-totp-code-input-label-v2',
+    inputLabelText: 'Enter 6-digit code',
+    pattern: '[0-9]{6}',
+    maxLength: 6,
+    submitButtonFtlId: 'signin-totp-code-confirm-button',
+    submitButtonText: 'Confirm',
+  };
+
+  const onSubmit = () => {
+    if (!code) {
+      // TODO: Add l10n for this string
+      // Holding on l10n pending product decision
+      // Current string vs "Security code required" vs other
+      // See FXA-6422, and discussion on PR-14744
+      setCodeErrorMessage('Two-step authentication code required');
+    }
+    try {
+      // Check security code
+      // logViewEvent('flow', `${viewName}.submit`, ENTRYPOINT_REACT);
+      // redirect to /pair/auth/allow
+    } catch (e) {
+      // TODO: error handling, error message confirmation
+      //       - decide if alertBar or error div
+      // const errorAuthTotp = ftlMsgResolver.getMsg(
+      //   'signin-totp-code-error-general',
+      //   'Invalid confirmation code'
+      // );
+      // alertBar.error(errorAuthTotp);
+    }
+  };
+
+  return (
+    // TODO: redirect to force_auth or signin if user has not initiated sign in
+    <>
+      {/* Ftl ids match those in signin-totp-code because it uses those strings. */}
+      <CardHeader
+        headingWithDefaultServiceFtlId="signin-totp-code-heading-w-default-service"
```

Comment:
Wondering if we should create new FTL strings for separation of concerns (event if the strings are currently identical), to avoid side-effects if the strings get changed for either page in the future. As it stands, it's not clear from the `SigninTotpCode` page that changing strings there will impact this page too.

Reply:
I can split them out if you think that's better!

Reply:
I'll add a comment to the `SigninTotpCode` view to make it clear that they're duplicated, so that we can split them out if necessary